### PR TITLE
dom/met signature: `make_base_ptr` -> `make_base_laplace_threshold`

### LINKED ISSUE #435: `make_base_ptr` revisions
1. `ptr` is too general a name to be used for only stability histogram releases.
3. Change to dom/met convention
4. replace ε(δ) curve with (ε, δ)
5. improve bounds checking on arguments

### PR DESCRIPTION
**Pull request by Shoeboxam**
Closes #435

1. `ptr` is too general a name to be used for only stability histogram releases.
3. Change to dom/met convention
4. replace ε(δ) curve with (ε, δ)
5. improve bounds checking on arguments
6. improve floating-point protections

### CODE REVIEW

**Comment by Shoeboxam** [comment]
Current dependencies on/for this PR:
* main
  * **PR #849** <a href="https://app.graphite.dev/github/pr/opendp/opendp/849" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/opendp/opendp/849?utm_source=stack-comment).

File: rust/src/measurements/laplace_threshold/ffi.rs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        make_base_laplace_threshold::<TK, TV>(input_domain, input_metric, scale, threshold, Some(k)).into_any()
+    }
+    let k = k as i32;
+
+    let input_domain = try_as_ref!(input_domain);
+    let input_metric = try_as_ref!(input_metric);
+    let TypeContents::GENERIC {name, args} = &input_domain.carrier_type.contents else {
+        return err!(FFI, "Generic type {:?} not supported", input_domain.type_.descriptor).into();
+    };
+    if name.starts_with("MapDomain") || args.len() != 2 {
```
Comment (andrewvyrros) [inline rust/src/measurements/laplace_threshold/ffi.rs:46]: Shouldn't this be `!name.starts_with("MapDomain")`? But then if so, how are the tests passing?

**Review by andrewvyrros — COMMENTED** [review]
I like this repositioning!

File: rust/src/measurements/laplace_threshold/ffi.rs (hunk 1 above)
Comment (Shoeboxam) [reply rust/src/measurements/laplace_threshold/ffi.rs:46]: Right, but it should be `!name.starts_with("HashMap")`

**Review by andrewvyrros — COMMENTED** [review]
Looks good (pending rebase)!

**Review by andrewvyrros — APPROVED** [review]

**Pull request merged**